dependencies.yaml

```diff
@@ -0,0 +1,83 @@
+# Dependency list for https://github.com/rapidsai/dependency-file-generator
+files:
+  all:
+    output: none
+    includes:
+      - checks
+      - cudatoolkit
+      - py_version
+      - run
+      - test_python
+  test_python:
+    output: none
+    includes:
+      - cudatoolkit
+      - py_version
+      - test_python
+  checks:
+    output: none
+    includes:
+      - checks
+      - py_version
+channels:
+  - rapidsai
+  - rapidsai-nightly
+  - dask/label/dev
+  - conda-forge
+  - nvidia
```

Comment:
Are these channels here for testing as well, or just for building/distributing? In the latter case both `dask/label/dev` and `nvidia` can be removed, as UCX-Py doesn't (or shouldn't) have any hard dependencies on either.

Reply:
These channels are used when we generated the `environment.yaml` files used for our test environments in CI. (e.g. [here](https://github.com/rapidsai/ucx-py/actions/runs/3914114616/jobs/6691144626#step:6:18)).

I just copy/pasted the default channels we use from other libraries but I can remove `dask` and `nvidia` if they're extraneous.

Reply:
The reason I commented is really because I don't know exactly what they're used for. If they're not going to pull any potential hard dependencies into a build (which probably won't happen indeed) then we can keep them. IOW, users may not want CUDA, RAPIDS or Dask at all, and thus we don't want to introduce any such dependencies.

Reply:
Gotcha. Yeah, they're just used for the files that get generated by the new [rapids-dependency-file-generator](https://github.com/rapidsai/dependency-file-generator) tool and `dependencies.yaml` file.

For example, here's an environment file in `cudf` that was generated by the tool:

- https://github.com/rapidsai/cudf/blob/ec7f8c6b070e421f4efa748a2da08465651701aa/conda/environments/all_cuda-118_arch-x86_64.yaml#L1-L9

The tool is also used to generate testing environments for CI (e.g. [here](https://github.com/ajschmidt8/ucx-py/blob/55c8cb02c48d567506e7250471a2e19e722f899e/ci/test_python.sh#L8-L14) and corresponding [log output](https://github.com/rapidsai/ucx-py/actions/runs/3914114616/jobs/6691144626#step:6:18)).

This lets us avoid having to repeat dependencies in several areas.

In the long term, we'd like to continue iterating on the tool so that `dependencies.yaml` can be a single source of truth for all dependencies of a given repository, but for now it does just enough to support keeping things DRY for GH Actions.